addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -117,6 +117,10 @@ def action_confirm(self):
     def button_validate(self):
         res = super().button_validate()
         to_assign_ids = set()
+        # Having non-done pickings after the `super()` call means it stopped early,
+        # so we shouldn’t remove the pickings from batches yet.
+        if not any(picking.state == 'done' for picking in self):
```

Comment:
I would prefer checking `res` is an action or not like here 

https://github.com/odoo/odoo/blob/82eaa0aa4b327054f3e2f992385458a81d55676a/addons/stock/models/stock_picking.py#L1145-L1147

Reply:
@Whenrow thanks for your review but we already tried to address this in a previous fix:
https://github.com/odoo/odoo/commit/838718cc7631d1bdd5a7de4f398f8c7b2ac366b0